common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp

```diff
@@ -451,7 +451,7 @@ static bool get_scan_results_from_nl_msg(sChannelScanResults &results, struct nl
 /////////////////////////////// Implementation ///////////////////////////////
 //////////////////////////////////////////////////////////////////////////////
 
-mon_wlan_hal_dwpal::mon_wlan_hal_dwpal(std::string iface_name, hal_event_cb_t callback)
+mon_wlan_hal_dwpal::mon_wlan_hal_dwpal(const std::string &iface_name, hal_event_cb_t callback)
```

Comment:
This constructor just as an example, this comment is for all changes that replaced a copy with a const reference:
are we sure the const reference we pass does not go out of scope leaving us with a dangling reference?
If we bind to a temporary, its lifetime is extended. But if we bind to a variable and this variable goes out of scope, then we have a problem.
reference: https://en.cppreference.com/w/cpp/language/reference#Dangling_references

Reply:
I have to check it.
I hope there is no constructor/method that stores references to `std::string`s instead of copying them.

Reply:
In fact, if it was a reference to a std::string, it wouldn't have worked to begin with, because the temporary that was passed originally goes out of scope much quicker and it *will* be overwritten with something else on the stack. So if a reference was stored in the object, we'd already have noticed.